plugins/chat/assets/javascripts/discourse/lib/transform-auto-links.js

```diff
@@ -15,6 +16,18 @@ function transform(html) {
     mentionLink.classList.add("mention");
     mentionLink.appendChild(mentionText);
     mentionLink.href = getURL(`/u/${mentionSpan.innerText.substring(1)}`);
-    mentionSpan.parentNode.replaceChild(mentionLink, mentionSpan);
+    mentionSpan.replaceWith(mentionLink);
+  });
+
+  (html.querySelectorAll("span.hashtag-raw") || []).forEach((hashtagSpan) => {
+    // Doesn't matter what "type" of hashtag we use here, it will get replaced anyway,
+    // this is just for the placeholder HTML.
+    generatePlaceholderHashtagHTML("category", hashtagSpan, {
+      id: -1,
+      text: "...",
+      relative_url: "/",
+      slug: "",
+      icon: "square-full",
+    });
```

Comment:
Now, this function does two operations, and readers will have to scan the functions' implementation in order to figure out that. We can make it immediately obvious (and also prevent the parts from coupling with each other) by stating explicitly what's going on:

```
function transform(html) {
  transformMentions(html);
  transformHashtags(html);
}
```

Or even better, we can refactor the `transformAutolinks` function:
```
  export default function transformAutolinks(cooked) {
    const html = domParser.parseFromString(cooked, "text/html");
    transformMentions(html);
    transformHashtags(html);
    return html.body.innerHTML;
  }
```